Time and Time Again/src/TimeAndTimeAgain.java

```diff
@@ -0,0 +1,40 @@
+import java.io.File;
+import java.util.Scanner;
+import java.util.regex.Pattern;
+import java.util.regex.Matcher;
+import java.io.File;
+import java.io.IOException;
+
+public class TimeAndTimeAgain
+	{
+		public static void main(String[] args) throws IOException
+			{
+			Scanner input = new Scanner(new File("TimeAndTimeAgainTextfile.txt"));
+			int numberOfCases = input.nextInt();
+			for (int i = 0; i < numberOfCases; i++)
+				{
+				int hour = 0;
+				int minute = 0; 
+				int second = 0;
+				String data = input.nextLine();
+				
+				Pattern secondRegex = Pattern.compile(".*(\\d+)s.*");
```

Comment:
So it looks like only the second digit is ever being read. I checked this regex and you're missing a "?".  -
`Pattern secondRegex = Pattern.compile(".*?(\\d+)s.*")` when I added the ? into your code for the second/minute/hour patterns, it read the entire number.